### PR TITLE
Enable gofumpt

### PR DESCRIPTION
**Pull request by faddat**
## Description

PR aimed at enabling fumpt linter and using golangci-lint run --fix to make developer environments match ci

* adds `make tools`
* fixes `make format`

Should be merged after #1177 since this also updates to go 1.18

______

For contributor use:

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

**Comment by ValarDragon** [comment]
None of the changes gofumpt is making seem particularly helpful to me

File: Makefile [hunk 1]
```diff
@@ -231,12 +231,13 @@ benchmark:
 
 lint:
 	@echo "--> Running linter"
-	@go run github.com/golangci/golangci-lint/cmd/golangci-lint run --timeout=10m
+	goglangci-lint run --timeout=10m
```
Comment (alexanderbez) [inline Makefile:234]: Undo. See https://github.com/osmosis-labs/osmosis/pull/1133

Using the tools methodology, the correct version of the dep is automatically fetched allowing us to use what go mod gets, thus not having to require dev to keep their local deps in sync.

File: Makefile [hunk 2]
```diff
@@ -231,12 +231,13 @@ benchmark:
...
 
 format:
-	find . -name '*.go' -type f -not -path "./vendor*" -not -path "*.git*" -not -path "./client/lcd/statik/statik.go" | xargs gofmt -w -s
-	find . -name '*.go' -type f -not -path "./vendor*" -not -path "*.git*" -not -path "./client/lcd/statik/statik.go" | xargs misspell -w
-	find . -name '*.go' -type f -not -path "./vendor*" -not -path "*.git*" -not -path "./client/lcd/statik/statik.go" | xargs goimports -w -local github.com/cosmos/cosmos-sdk
+	find . -name '*.go' -type f -not -path "./vendor*" -not -path "*.git*" -not -path "./client/docs/statik/statik.go" -not -name "*.pb.gw.go" -not -name "*.pb.go" | xargs gofumpt -w -s
+	find . -name '*.go' -type f -not -path "./vendor*" -not -path "*.git*" -not -path "./client/docs/statik/statik.go" -not -name "*.pb.gw.go" -not -name "*.pb.go" | xargs misspell -w
+	find . -name '*.go' -type f -not -path "./vendor*" -not -path "*.git*" -not -path "./client/docs/statik/statik.go" -not -name "*.pb.gw.go" -not -name "*.pb.go" | xargs goimports -w -local github.com/cosmos/cosmos-sdk
+	golangci-lint run ./... --fix
```
Comment (alexanderbez) [inline Makefile:240]:
```suggestion
	@go run github.com/golangci/golangci-lint/cmd/golangci-lint run ./... --fix
```

File: Makefile [hunk 3]
```diff
@@ -246,3 +247,13 @@ format:
...
+	
+	
+###############################################################################
+###                                Tools                                    ###
+###############################################################################
+
+tools:
+	go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.45.0
+	go install mvdan.cc/gofumpt@latest
+
```
Comment (alexanderbez) [inline Makefile:259]:
```suggestion
```

Just add `_ mvdan.cc/gofumpt` to `tools.go`

**Review by alexanderbez — CHANGES_REQUESTED** [review]

File: Makefile (hunk 3 above)
Comment (faddat) [reply Makefile:259]: and now I get it......

OK :)

**Comment by faddat** [comment]
@alexanderbez if Dev would like fumpt I'll take care of the rest that you've requested here

**Comment by faddat** [comment]
@alexanderbez changes made :)

**Review by alexanderbez — APPROVED** [review]
There's no way for me to review the entire set of changes here, but I assume all the changes are to confirm to our linters and that there's no core business logic changes, so ACK.

**Review by ValarDragon — APPROVED** [review]
skimmed, everything seems like import changes

**Pull request merged**